apis/placement/v1alpha1/stagerollout_types.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Namespaced",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageRollout defines a stage by stage rollout policy that is applied to the ClusterResourcePlacement it tracks.
+type StageRollout struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of StageRollout.
+	// +required
+	Spec StageRolloutSpec `json:"spec"`
+
+	// The observed status of StageRollout.
+	// +optional
+	Status StageRolloutStatus `json:"status,omitempty"`
+}
+
+// StageRolloutSpec defines the desired the rollout sequence stage by stage.
+type StageRolloutSpec struct {
+	// The name of the CRP that this rollout is applied to.
+	// There can only be one active rollout for each ClusterResourcePlacement,
+	// we will fail the rollout if there is already an active rollout.
+	// +required
+	ParentCRPName string `json:"parentCRPName"`
```

Comment:
i'm thinking of the other way. CRP refers to the StageRollout name.
StageRollout will be a common rollout configurations for multiple CRP.


All the rollout status will be part of the CRP status, like today.